nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -1209,13 +1229,17 @@ def filter_conns_by_input_or_internal_constraints(self, conns1, conns2):
         """
         peers_to_compare = conns1.project_on_one_dimension('src_peers') | conns1.project_on_one_dimension('dst_peers') | \
             conns2.project_on_one_dimension('src_peers') | conns2.project_on_one_dimension('dst_peers')
-        exclude_ipv6 = self.output_config.excludeIPv6Range
-        ref_ip_blocks = self.config1.get_referenced_ip_blocks(exclude_ipv6) | \
-            self.config2.get_referenced_ip_blocks(exclude_ipv6)
-        ip_blocks_mask = IpBlock() if ref_ip_blocks else IpBlock.get_all_ips_block(exclude_ipv6)
-        for ip_block in ref_ip_blocks:
-            ip_blocks_mask |= ip_block
-        peers_to_compare.filter_ipv6_blocks(ip_blocks_mask)
+        exclude_ipv6 = self.config1.check_for_excluding_ipv6_addresses(self.output_config.excludeIPv6Range) and \
+            self.config2.check_for_excluding_ipv6_addresses(self.output_config.excludeIPv6Range)
+
+        ip_blocks_mask = IpBlock()
+        ipv6_full_block = IpBlock.get_all_ips_block(exclude_ipv4=True)
+        for ip_block in peers_to_compare:
+            if isinstance(ip_block, IpBlock) and (not exclude_ipv6 or ip_block != ipv6_full_block):
+                ip_blocks_mask |= ip_block
+        if not ip_blocks_mask:
+            ip_blocks_mask = IpBlock.get_all_ips_block(exclude_ipv6)
+        peers_to_compare.filter_ip_blocks_by_mask(ip_blocks_mask)
```

Comment:
need some more documentation on this function

Reply:
Simplified